shell/platform/fuchsia/flutter/engine.h

```diff
@@ -53,8 +53,8 @@ class Engine final {
  private:
   Delegate& delegate_;
   const std::string thread_label_;
+  flutter::ThreadHost thread_host_;
```

Comment:
This seems like a strange class to own the thread host, but that's how we had it before I guess.

Is this migration going to help us migrate Fuchsia to the embedder API eventually, or is it completely unrelated?

@cbracken

Reply:
It should do. The `ThreadHost` holds the various `fml::Thread` objects as `std::unique_ptr`s which will get destroyed when the `ThreadHost` goes out of scope, and in their dtor they call `Thread::Join()` which in turn terminates the `MessageLoop`: https://github.com/flutter/engine/blob/master/fml/thread.cc#L41

Reply:
This change should definitely move us in the direction of migrating Fuchsia to using the embedder API.